server/src/internalClusterTest/java/org/opensearch/remotestore/CreateRemoteIndexTranslogDisabledIT.java

```diff
@@ -54,10 +53,9 @@ public void testRemoteStoreEnabledByUserWithRemoteRepo() throws Exception {
     }
 
     public void testDefaultRemoteStoreNoUserOverride() throws Exception {
-        final int numReplicas = internalCluster().numDataNodes();
         Settings settings = Settings.builder()
-            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 3)
-            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, numReplicas)
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
```

Comment:
why were the tests flaky ?

Reply:
+1. Also how have you determined that they are no longer flaky?

Reply:
The failures were due to the following error:

```
java.lang.AssertionError: [test-idx-1][2], node[vX6f0DTmRfC1il8YDrSIkQ], [P], s[STARTED], a[id=53rPm8H3SrepttaigUli0A] has unreleased snapshotted index commits

java.lang.RuntimeException: file handle leaks: [InputStream(/var/jenkins/workspace/gradle-check/search/server/build/testrun/internalClusterTest/temp/org.opensearch.remotestore.CreateRemoteIndexIT_58A3D614EB68203F-001/tempDir-008/repos/fgUzGkzgHm/qWmYzVJSTKeMhahfEsSuhw/0/segments/data/segment_infos_snapshot_filename__3__PXCpdocBnatMeQEWYB27)]
```

https://build.ci.opensearch.org/job/gradle-check/13937/testReport/junit/org.opensearch.remotestore/CreateRemoteIndexIT/testRemoteStoreDisabledByUser/

I was able to repro with following params:

```
./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.CreateRemoteIndexIT" -Dtests.seed=58A3D614EB68203F -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en -Dtests.timezone=Etc/UTC -Dtests.iters=20

./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.CreateRemoteIndexTranslogDisabledIT" -Dtests.seed=58A3D614EB68203F -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en -Dtests.timezone=Etc/UTC -Dtests.iters=20

./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.CreateRemoteIndexClusterDefaultDocRep" -Dtests.seed=58A3D614EB68203F -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en -Dtests.timezone=Etc/UTC -Dtests.iters=20
```

Mostly it was because we were not cleaning up snapshot dir. 

After my changes, i've ran above 3 commands multiple times and havent seen any failures.

changes around replica count and shard count are done just to simplify the tests further, our main goal with these tests is to verify cluster level settings behavior with remote store